### PR TITLE
Fix output option for stdout (fixes #62)

### LINKED ISSUE #62: Specifying dash for output file no longer works
After reading your `README`, I realize that my latest changes break the `-o-` option. Instead of outputting to stdout, it outputs to a file called `-`. I'll be working on a PR for this soon. Sorry for the inconvenience.

### PR DESCRIPTION
**Pull request by rzuckerm**
* PR #61 broke the ability to specify stdout for `-o/--output`. This PR fixes that
* Added tests for `-o/--output` for stdout

### CODE REVIEW

File: compiledb/__init__.py [hunk 1]
```diff
@@ -37,16 +41,24 @@ def generate_json_compdb(instream=None, proj_dir=os.getcwd(), verbose=False, exc
 
 def write_json_compdb(compdb, compdb_path='compile_commands.json', verbose=False,
                       force=False, pretty_output=True):
-    print("## Writing compilation database with {} entries to {}".format(
-        len(compdb), compdb_path))
+    if not __is_stdout(compdb_path):
```
Comment (rzuckerm) [inline compiledb/__init__.py]: Added separate logic for stdout. This turned out to be a lot less code than doing something like this [Stack Overflow article](https://stackoverflow.com/questions/17602878/how-to-handle-both-with-open-and-sys-stdout-nicely)

File: compiledb/__init__.py [hunk 2]
```diff
@@ -37,16 +41,24 @@ def generate_json_compdb(instream=None, proj_dir=os.getcwd(), verbose=False, exc
...
+            outstream.write(os.linesep)
+    else:
+        print("## Writing compilation database with {} entries to <stdout>".format(
+            len(compdb)))
+        print(json.dumps(compdb, indent=pretty_output))
 
 
 def load_json_compdb(compdb_path='compile_commands.json', verbose=False):
     try:
+        if __is_stdout(compdb_path):
```
Comment (rzuckerm) [inline compiledb/__init__.py]: Don't try to read stdout

File: tests/test_json_compdb.py [hunk 3]
```diff
@@ -49,6 +48,7 @@
     }
 ]
 nonexistent_files = ['compile_commands.json', 'nonexistent.json']
+stdout_filenames = ['-', '']
```
Comment (rzuckerm) [inline tests/test_json_compdb.py]: Added new tests for stdout and refactored existing tests:
- Renamed `assert_compdb_equals` function to `assert_compdb_file_equals`
- Extracted compilation database compare to separate `assert_compdb_equals` function

File: tests/test_json_compdb.py [hunk 4]
```diff
@@ -181,14 +207,21 @@ def assert_generate_is_true(output_filename, overwrite):
             strict=False
         )
 
-def assert_compdb_equals(compdb_path, expected_compdb):
+def assert_compdb_file_equals(compdb_path, expected_compdb):
     def get_key(item):
```
Comment (rzuckerm) [inline tests/test_json_compdb.py]: Oops, need to remove this function. Will fix on next commit

**Comment by agnjunio** [comment]
I think this feature could be more effective with a silent mode default for stdout outputs, but this works fine.

File: compiledb/__init__.py (hunk 2 above)
Comment (agnjunio) [reply compiledb/__init__.py]: compiledb supports stdin input, wouldn't make sense to try to load from it?

**Comment by nickdiego** [comment]
I think we can improve this.
Could you try to revert to use [Click.File arguments](https://click.palletsprojects.com/en/7.x/arguments/#file-arguments) for output, but opening it for reading+writing ([trivial example here](https://stackoverflow.com/questions/6648493/how-to-open-a-file-for-both-reading-and-writing)), this way we could use direct comparison against `sys.stdout` to avoid trying to load compdb from `stdout`, what IMHO would be way more elegant and readable.

**Comment by rzuckerm** [comment]
@nickdiego My first approach was to try `click.File('rw')`, but that didn't work if the file didn't exist. Can this PR be merged? Thanks!

**Comment by nickdiego** [comment]
Thanks for contributing @rzuckerm and sorry for the delay.

**Pull request merged**